src/editors/playgroundController.ts

```diff
@@ -171,13 +176,67 @@ export default class PlaygroundController {
     });
   }
 
-  public showActiveConnectionInPlayground(message: string): Promise<boolean> {
-    return new Promise((resolve) => {
-      this._outputChannel.clear();
-      this._outputChannel.append(message);
-      this._outputChannel.show(true);
+  private getСonnectionQuickPicks(): any[] {
+    if (!this._connectionController._connections) {
+      return [
+        {
+          label: 'Add new connection',
+          data: {
+            type: NewConnectionType.NEW_CONNECTION
+          }
+        }
+      ];
+    }
+
+    return [
+      {
+        label: 'Add new connection',
+        data: {
+          type: NewConnectionType.NEW_CONNECTION
+        }
+      },
+      ...Object.values(this._connectionController._connections)
+        .map((item) => ({
+          label: item.name,
+          data: {
+            type: NewConnectionType.SAVED_CONNECTION,
+            connectionModel: item.connectionModel
+          }
+        }))
+        .sort((collectionA: any, collectionB: any) => {
+          if (collectionA.name < collectionB.name) {
+            return -1;
+          }
+          if (collectionA.name > collectionB.name) {
+            return 1;
+          }
+          return 0;
+        })
+    ];
+  }
 
-      resolve(true);
+  public changeActiveConnection(): Promise<boolean> {
+    return new Promise(async (resolve) => {
+      const selectedQuickPickItem = await vscode.window.showQuickPick(
+        this.getСonnectionQuickPicks(),
+        {
+          placeHolder: 'Select new connection...'
+        }
+      );
+
+      if (!selectedQuickPickItem) {
+        return resolve(true);
+      }
+
+      if (
+        selectedQuickPickItem.data.type === NewConnectionType.NEW_CONNECTION
+      ) {
+        return this._connectionController.connectWithURI();
+      }
+
+      return this._connectionController.parseNewConnectionAndConnect(
+        selectedQuickPickItem.data.connectionModel
```

Comment:
Looks like this creates a new connection with each connection. I think when the user clicks on an existing connection we want to connect to that connection and not recreate a saved connection.

Maybe we can use `connectWithConnectionId` on the `connectionController` instead?

```js
return this._connectionController.parseNewConnectionAndConnect(
  selectedQuickPickItem.data.connectionId
);
```

and above we can change the `data` property to `connectionId` instead of `connectionModel`.